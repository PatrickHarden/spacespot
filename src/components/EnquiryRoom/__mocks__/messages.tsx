export const messages = [
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message',
    sentAt: '2018-10-18T12:40:35.886+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message1',
    sentAt: '2018-10-18T12:40:43.547+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message2',
    sentAt: '2018-10-18T12:40:44.946+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message3',
    sentAt: '2018-10-18T12:40:45.678+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message4',
    sentAt: '2018-10-18T12:40:46.856+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message5',
    sentAt: '2018-10-23T14:17:59.662+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'jk',
    sentAt: '2018-10-28T10:13:59.805+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'message6',
    sentAt: '2018-10-28T10:16:16.585+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '11:44',
    sentAt: '2018-10-28T10:45:02.592+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:34',
    sentAt: '2018-10-28T11:23:27.467+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'll',
    sentAt: '2018-10-28T11:29:37.057+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: '12:30',
    sentAt: '2018-10-28T11:30:13.643+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'The message',
    sentAt: '2018-10-29T08:51:21.007+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Another',
    sentAt: '2018-10-29T09:28:49.589+0000',
  },
  {
    userId: 'f9251bff-4527-4ebd-9028-f77b4bd0e0ce',
    message: 'Other',
    sentAt: '2018-10-29T09:29:02.900+0000',
  },
]
