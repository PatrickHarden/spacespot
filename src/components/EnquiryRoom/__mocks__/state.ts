export const mockState: any = {
  enquiry: {
    data: {
      spaceId: 'SS_SPACE_665',
      peer1Id: '1',
      peer2Id: '2',
      chatId: 159,
      enquireStatus: 'ACCEPTED',
      negotiationId: 139,
      enquireId: 159,
      flex: false,
    },
  },
  negotiation: {
    data: {
      flexible: false,
      terms: [
        {
          termId: 5,
          value: 'MONTHLY',
        },
        {
          termId: 6,
          value: '2000',
        },
        {
          termId: 7,
          value: '2',
        },
        {
          termId: 8,
          value: '2000',
        },
        {
          termId: 9,
          value: 'GUARANTEE',
        },
        {
          termId: 1,
          value: '3500',
        },
        {
          termId: 2,
          value: '40000',
        },
        {
          termId: 3,
          value: '2020-04-22',
        },
        {
          termId: 4,
          value: '13',
        },
      ],
      negotiationId: 139,
      enquireId: 159,
      createdAt: '2020-02-14T06:30:08.354+0000',
      lastUpdated: '2020-03-05T16:35:02.513+0000',
      lastModifyBy: '0062fb00-e389-4bb8-9119-613e3c45938a',
      status: 'PROPOSED',
      customFitout: false,
      fitoutOption: 0,
      specialProvision: 'gfd',
      specialProvisionStatus: 'CREATED',
    },
    loading: false,
    termsFields: {
      '1': 'Area',
      '2': 'Rent',
      '3': 'Start',
      '4': 'Duration',
      '5': 'Frequency',
      '6': 'AdditionalCosts',
      '7': 'Deposit',
      '8': 'ServiceCharges',
      '9': 'DepositType',
    },
    signerInfo: {
      id: 1,
      name: 'Name1',
      emailId: 'name1@email.com',
      companyName: 'companyA',
      companyNumber: '12345678',
    },
    customFitout: [],
  },
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
