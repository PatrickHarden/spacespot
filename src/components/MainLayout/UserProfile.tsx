import React, { useState, Dispatch, SetStateAction } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { useIntl } from 'react-intl'
import styled from 'styled-components'

import Popover from '@material-ui/core/Popover'
import enquiryActions from 'services/myEnquiries/actions'

import selectors from 'services/user/selectors'
import actions from 'services/user/actions'
import { Link } from 'react-router-dom'
import dashboardActions from 'services/dashboard/actions'
import { useNotificationCount } from 'services/myEnquiries/hooks'

import Avatar from 'components/Avatar/Avatar'
import FilledButton from 'components/common/FilledButton'
import device from 'services/global/device'
import Close from '@material-ui/icons/Close'
import useMediaQuery from '@material-ui/core/useMediaQuery'
import EditProfile from './EditProfile'
import { useProfile } from 'services/user/hooks'
import company from 'assets/icons/company.svg'
import enquiries from 'assets/icons/enquiries.svg'
import analytics from 'services/analytics'

const CloseIcon = styled(Close)`
  display: none;
  position: absolute;
  top: 5px;
  right: 10px;
`
const ContainerAvatar = styled.div`
  flex: 0 0 54px;
  padding: 10px 20px 0 0;

  @media ${device.mobile} {
    padding: 20px 30px 0 0;
  }
`
const LinkBadge = styled.div`
  background-color: #db4437;
  text-align: center;
  border-radius: 50%;
  color: white;
  padding: 5px 6px;
  line-height: 9px;
  font-size: 11px;
  margin-left: 10px;
`
const Img = styled.img`
  height: 30px;
`
const Text = styled.span`
  display: inline-block;
`
const UserProfileContentContainer = styled.div`
  position: relative;
  width: 360px;
  padding: 10px;
  @media (max-width: 480px) {
    padding: 0;
    width: 100vw;
  }
`
const ContainerFlex = styled.div`
  display: flex;
  border-bottom: 1px solid #ddd;
  margin: 0 10px;

  @media ${device.mobile} {
    margin: 0 20px;
    padding-top: 10px;
  }
`
const ContainerInfo = styled.div`
  padding: 10px 0 5px;

  @media ${device.mobile} {
    padding: 20px 0 5px;
  }
`
const Name = styled.div`
  color: #000000;
  font-size: 18px;
  font-weight: bold;
`
const Email = styled.div`
  color: #828286;
  font-size: 16px;
  font-weight: 400;
  padding: 5px 0 10px 0;
`
const UserProfileButton = styled(FilledButton)`
  color: #f35c2b;
  background: transparent;
  padding: 0;
  outline: 0;
`
const SignOut = styled.button`
  display: block;
  color: #ffffff;
  font-size: 16px;
  font-weight: 500;
  margin: 20px auto 10px;
  padding-top: 0;
  border: none;
  cursor: pointer;
  background: #f35c2b;
  padding: 8px 40px;
  border-radius: 4px;
  outline: none;

  @media ${device.mobile} {
    margin: 20px auto;
  }
`

const LinkItem = styled.li`
  white-space: nowrap;
  margin: 0 20px;
  font-size: 16px;
  font-weight: 500;
  list-style: none;
  cursor: pointer;
  border-bottom: 1px solid #dddddd;

  a {
    display: flex;
    align-items: center;
    padding: 15px 10px 15px 104px;
    border: 2px solid transparent;
    text-decoration: none;
    transition: 0.3s;
    color: #404042;
    outline: none;
    font-size: 17px;
    font-weight: normal;

    @media ${device.mobile} {
      padding: 15px 10px 15px 80px;
    }

    img {
      margin-right: 20px;
      width: 32px;
    }
    :hover {
      color: #404042;
      border-bottom: 2px solid white;
    }
    :focus {
      color: #404042;
      border-bottom: 2px solid white;
    }
  }
`
const LinksList = styled.ul`
  margin: 0;
  padding: 0;
`
const UserProfileContent = (props: {
  setShow: Dispatch<SetStateAction<boolean>>
}) => {
  const { setShow } = props
  const dispatch = useDispatch()
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const isLandlord = useSelector(selectors.isLandlord)
  const ownUserId = useSelector(selectors.accountIdentifier)
  const profile = useProfile(ownUserId)
  const userName = profile ? profile.displayName : ''
  const userEmail = profile ? profile.emailId : ''
  const [edit, setEdit] = useState(false)
  const logout = () => {
    dispatch(actions.logout())
  }

  const isMobile = useMediaQuery(device.mobile)
  const notifications = useNotificationCount()
  analytics.gtmEvent(useSelector(selectors.getGAUser))

  const eventClickGa = (action: string) =>
    analytics.event({
      category: 'Header navigation',
      action,
    })

  const refresh = () => {
    dispatch(dashboardActions.dashboardInit())
    eventClickGa('My spaces')
  }

  const refreshEnquiries = () => {
    dispatch(enquiryActions.get())
    eventClickGa('Enquiries')
  }

  if (edit) {
    return (
      <UserProfileContentContainer>
        <CloseIcon onClick={() => setShow(false)} />
        <EditProfile setShow={setEdit} />
      </UserProfileContentContainer>
    )
  }
  return (
    <UserProfileContentContainer>
      <CloseIcon onClick={() => setShow(false)} />
      <ContainerFlex>
        <ContainerAvatar>
          <Avatar isLandlord={isLandlord} userId={ownUserId} />
        </ContainerAvatar>
        <ContainerInfo>
          <Name>{userName}</Name>
          <Email>{userEmail}</Email>
          <UserProfileButton onClick={() => setEdit(true)}>
            {t('NAV_EDIT_PROFILE')}
          </UserProfileButton>
        </ContainerInfo>
      </ContainerFlex>
      {isMobile && (
        <LinksList>
          <LinkItem data-testid="enquiries">
            <Link to="/enquiries" onClick={refreshEnquiries}>
              <Img src={enquiries} alt="enquiries" />
              <Text>{t('NAV_ENQUIRIES')}</Text>
              {notifications > 0 && <LinkBadge>{notifications}</LinkBadge>}
            </Link>
          </LinkItem>
          {isLandlord && (
            <LinkItem data-testid="dashboard">
              <Link to="/dashboard" onClick={refresh}>
                <Img src={company} alt="spaces" />
                <Text>{t('NAV_SPACES')}</Text>
              </Link>
            </LinkItem>
          )}
        </LinksList>
      )}
      <SignOut data-testid="logout" onClick={logout}>
        {t('NAV_SIGN_OUT')}
      </SignOut>
    </UserProfileContentContainer>
  )
}

const UserProfile = (props: {
  setShow: Dispatch<SetStateAction<boolean>>
  show: boolean
  anchorEl?: Element | null
}) => {
  const { setShow, show, anchorEl } = props

  return (
    <Popover
      id="userpopover"
      open={show}
      onClose={() => setShow(false)}
      anchorEl={anchorEl}
      anchorOrigin={{
        vertical: 'bottom',
        horizontal: 'center',
      }}
      transformOrigin={{
        vertical: 'top',
        horizontal: 'center',
      }}>
      <UserProfileContent setShow={setShow} />
    </Popover>
  )
}
export default UserProfile
