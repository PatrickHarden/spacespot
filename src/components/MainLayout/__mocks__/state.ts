export const mockState: any = {
  user: {
    auth: {
      idToken: {
        claims: {
          // eslint-disable-next-line @typescript-eslint/camelcase
          extension_Landlord: true,
        },
      },
    },
  },
}
