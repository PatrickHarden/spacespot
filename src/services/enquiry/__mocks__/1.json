{
  "chatId": 1,
  "enquireId": 1,
  "spaceId": 2,
  "peer1Id": "8fa47f17-c44e-48eb-8e88-9e8c8fe5cda3",
  "peer2Id": "8fa47f17-c44e-48eb-8e88-9e8c8fe5cda4"
}
