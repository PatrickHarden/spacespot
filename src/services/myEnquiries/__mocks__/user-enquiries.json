[
  {
    "enquireId": 167,
    "negotiationId": 147,
    "enquireStatus": "PENDING_TO_ACCEPT",
    "spaceId": "SS_SPACE_650",
    "terms": [
      { "termId": 1, "value": "12" },
      { "termId": 2, "value": "12.0" },
      {
        "termId": 3,
        "value": "Tue Feb 25 2020 17:59:47 GMT+0100 (hora estándar de Europa central)"
      },
      { "termId": 4, "value": "123" },
      { "termId": 5, "value": "" },
      { "termId": 6, "value": "" },
      { "termId": 7, "value": "" },
      { "termId": 8, "value": "123.0" },
      { "termId": 9, "value": "" }
    ],
    "messages": [
      {
        "userId": "de2563f1-2e15-4a6a-9e39-9600a8ad4ec1",
        "message": "",
        "sentAt": "2020-02-25T16:59:53.122+0000"
      }
    ]
  },
  {
    "enquireId": 168,
    "negotiationId": 148,
    "enquireStatus": "PENDING_TO_ACCEPT",
    "spaceId": "SS_SPACE_650",
    "terms": [
      { "termId": 1, "value": "12" },
      { "termId": 2, "value": "12.0" },
      {
        "termId": 3,
        "value": "Tue Feb 25 2020 18:00:28 GMT+0100 (hora estándar de Europa central)"
      },
      { "termId": 4, "value": "13" },
      { "termId": 5, "value": "" },
      { "termId": 6, "value": "" },
      { "termId": 7, "value": "" },
      { "termId": 8, "value": "123.0" },
      { "termId": 9, "value": "" }
    ],
    "messages": [
      {
        "userId": "de2563f1-2e15-4a6a-9e39-9600a8ad4ec1",
        "message": "",
        "sentAt": "2020-02-25T17:00:34.014+0000"
      }
    ]
  }
]
