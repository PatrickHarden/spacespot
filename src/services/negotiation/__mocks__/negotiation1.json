{
  "terms": [
    { "termId": 1, "value": "34" },
    { "termId": 2, "value": "207" },
    { "termId": 3, "value": "29 Months" },
    { "termId": 4, "value": "200 Sq Mtr" },
    { "termId": 5, "value": "$5600 per month" },
    { "termId": 6, "value": "20th September" }
  ],
  "negotiationId": 129,
  "enquireId": 149,
  "createdAt": "2020-02-03T13:03:07.661+0000",
  "lastUpdated": "2020-02-11T09:06:15.764+0000",
  "lastModifyBy": "c700793a-1963-4b2b-996f-1939f853cc10",
  "status": "CREATED"
}
