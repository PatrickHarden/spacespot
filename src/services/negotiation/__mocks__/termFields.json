{
  "1": "Area",
  "2": "Rent",
  "3": "Start",
  "4": "Duration",
  "5": "Frequency",
  "6": "AdditionalCosts",
  "7": "Deposit",
  "8": "ServiceCharges"
}
