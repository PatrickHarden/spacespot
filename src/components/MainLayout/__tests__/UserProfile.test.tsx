import React, { useState } from 'react'
import { render, fireEvent } from '@testing-library/react'
import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'

import UserProfile from '../UserProfile'
import { mockState } from '../__mocks__/state'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
  createIntlCache: () => jest.fn(),
  createIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDispatch = jest.fn()

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}))

const Comp = () => {
  const [show, setShow] = useState(false)
  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(null)

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget)
    setShow(true)
  }
  return (
    <div>
      <button onClick={handleClick}>Click</button>
      <UserProfile show={show} setShow={setShow} anchorEl={anchorEl} />
    </div>
  )
}

it('Renders without crashing', () => {
  const store: any = createMockStore(mockState)
  const wrapper = render(
    <Provider store={store}>
      <Comp />
    </Provider>,
  )
  fireEvent.click(wrapper.getByText('Click'))
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.click(wrapper.getByText('NAV_EDIT_PROFILE'))
  expect(wrapper.asFragment()).toMatchSnapshot()
  fireEvent.change(wrapper.getByTestId('name'), {
    target: { value: 'UserName1' },
  })
  fireEvent.click(wrapper.getByText('EDIT_PROFILE_CANCEL'))
  fireEvent.click(wrapper.getByText('NAV_EDIT_PROFILE'))
  fireEvent.click(wrapper.getByText('EDIT_PROFILE_SAVE'))
})
