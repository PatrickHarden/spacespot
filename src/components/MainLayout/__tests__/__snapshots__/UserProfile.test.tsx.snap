// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <button>
      Click
    </button>
  </div>
</DocumentFragment>
`;

exports[`Renders without crashing 2`] = `
<DocumentFragment>
  <div>
    <button>
      Click
    </button>
  </div>
</DocumentFragment>
`;
